core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala

```diff
@@ -181,21 +181,26 @@ private[spark] object ReliableCheckpointRDD extends Logging {
       serializeStream.writeAll(iterator)
     } {
       serializeStream.close()
+      fileOutputStream.close()
     }
 
-    if (!fs.rename(tempOutputPath, finalOutputPath)) {
-      if (!fs.exists(finalOutputPath)) {
-        logInfo(s"Deleting tempOutputPath $tempOutputPath")
-        fs.delete(tempOutputPath, false)
-        throw new IOException("Checkpoint failed: failed to save output of task: " +
-          s"${ctx.attemptNumber()} and final output path does not exist: $finalOutputPath")
-      } else {
-        // Some other copy of this task must've finished before us and renamed it
-        logInfo(s"Final output path $finalOutputPath already exists; not overwriting it")
-        if (!fs.delete(tempOutputPath, false)) {
-          logWarning(s"Error deleting ${tempOutputPath}")
+    try {
```

Comment:
Agreed with @srowen , `FileSystem` is a cached object, closing it means removed it from cache. I don't think we need to call this explicitly. Because by default it is designed to be shared.